packages/mobile/src/recipients/reducer.ts

```diff
@@ -12,44 +12,54 @@ export interface State {
   // valoraRecipientCache contains accounts that the user has sent/recieved transactions from,
   // and includes CIP8 profile data if available
   valoraRecipientCache: AddressToRecipient
+  rewardsSenders: string[]
 }
 
-const initialState = {
+const initialState: State = {
   phoneRecipientCache: {},
   valoraRecipientCache: {},
+  rewardsSenders: [],
 }
 
-export const recipientsReducer = (
-  state: State | undefined = initialState,
-  action: ActionTypes | RehydrateAction
-) => {
-  switch (action.type) {
-    case REHYDRATE: {
+const rehydrate = createAction<any>(REHYDRATE)
+export const setPhoneRecipientCache = createAction<NumberToRecipient>(
+  'RECIPIENTS/SET_PHONE_RECIPIENT_CACHE'
+)
+export const updateValoraRecipientCache = createAction<AddressToRecipient>(
+  'RECIPIENTS/SET_VALORA_RECIPIENT_CACHE'
+)
+export const setRewardsSenders = createAction<string[]>('RECIPIENTS/SET_REWARDS_SENDERS')
+
+export const recipientsReducer = createReducer(initialState, (builder) => {
+  builder
+    .addCase(rehydrate, (state, action) => {
+      // hack to allow rehydrate actions here
+      const hydrated = getRehydratePayload((action as unknown) as RehydrateAction, 'recipients')
       return {
         ...state,
-        ...getRehydratePayload(action, 'recipients'),
+        ...hydrated,
         phoneRecipientCache: initialState.phoneRecipientCache,
       }
-    }
-    case Actions.SET_PHONE_RECIPIENT_CACHE:
-      return {
-        ...state,
-        phoneRecipientCache: action.recipients,
-      }
-    case Actions.UPDATE_VALORA_RECIPIENT_CACHE:
-      return {
-        ...state,
-        valoraRecipientCache: { ...state.valoraRecipientCache, ...action.recipients },
-      }
-    default:
-      return state
-  }
-}
+    })
+    .addCase(setPhoneRecipientCache, (state, action) => ({
+      ...state,
+      phoneRecipientCache: action.payload,
+    }))
+    .addCase(updateValoraRecipientCache, (state, action) => ({
+      ...state,
+      valoraRecipientCache: { ...state.valoraRecipientCache, ...action.payload },
+    }))
+    .addCase(setRewardsSenders, (state, action) => ({
+      ...state,
+      rewardsSenders: action.payload,
+    }))
+})
```

Comment:
Nice! I think we should go even further and use `createSlice` instead, but let's do this in separate PRs.
https://redux-toolkit.js.org/api/createslice